src/store/actions/auth-actions.js

```diff
@@ -0,0 +1,160 @@
+                    body 
+                });
+                responseData = await newResponse.json();
+
+            }
+            if (responseData.status >= 400 && responseData.status < 600) {
+                navigate('/sign');
+                return;
+            }
+
+            responseFn(responseData, dispatch);
+        } catch (e) {
+            console.log(e);
+            navigate('/sign');
+        }
+    };
+
+async function refreshAuthToken(headers) {
+    const refreshToken = cookies.get('Refresh');
+    if (!refreshToken) {
+        return null;
+    }
+
+    const refreshResponse = await fetch(urlEnum.refresh, {
+        method: 'GET',
+        headers: { ...headers, 'Authorization': refreshToken }
+    });
+
+    if (!refreshResponse) {
+        return null;
+    }
+    const refreshData = await refreshResponse.json();
+
+    if (refreshData.errorStatus) {
+        return null;
+    }
+
+    const { access_token, refresh_token } = refreshData.tokenPair;
+    cookies.set('Access', access_token);
+    cookies.set('Refresh', refresh_token);
+
+    return refreshData;
+}
+
+export const fetchRegister = (body, navigate) => {
+    return async (dispatch) => {
+        try {
+            body = JSON.stringify(body);
+            const response = await fetch(urlEnum.register, {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body
+            });
+
+            if (!response) {
+                throw new Error('Could not fetch data');
+            }
+
+            const data = await response.json();
+            if (data.status >= 400 && data.status < 600) {
+                throw new Error(data.message);
+            }
+
+            navigate('/profile');
+        } catch (e) {
+            return e;
+        }
+    }
+}
+
+export const fetchLogin = (body, navigate) => {
+    return async (dispatch) => {
+        try {
+            body = JSON.stringify(body);
+            const response = await fetch(urlEnum.login, {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body
+            });
+            console.log(body);
+
+            if (!response) {
+                throw new Error('Could not fetch data');
+            }
+
+            const data = await response.json();
+            if (data.status >= 400 && data.status < 600) {
+                throw new Error(data.message);
+            }
+
+            const { user, access_token, refresh_token } = data;
+
+            dispatch(authAction.updateAuth({
+                userInfo: { ...user, password: undefined },
+                userToken: access_token
+            }));
+
+            cookies.set('Access', access_token);
+            cookies.set('Refresh', refresh_token);
+
+            navigate('/profile');
+        } catch (e) {
+            console.log(e);
+            return e;
+        }
+    }
+}
+
+export const fetchUserInfo = () => {
```

Comment:
Пропоную зробити ось таку ф-ію , щоб прибрати повторення
```suggestion
const getFetch = (parameters, navigate, helpFn) => {
    return async (dispatch) => {
        try {
            const { url, method = 'GET', headers = {} } = parameters;

            const body = parameters.body ? JSON.stringify(parameters.body) : null;

            const response = await fetch(url, {
                method,
                headers,
                body
            });

            if (!response) {
                throw new Error('Could not fetch data');
            }

            const data = await response.json();
            if (data.status >= 400 && data.status < 600) {
                throw new Error(data.message);
            }

            if (helpFn) {
                helpFn(data, navigate, dispatch);
            }
        } catch (e) {
            console.log(e);
            return e;
        }
    }
}
```